packages/expo-router/src/global-state/router-store.tsx

```diff
@@ -143,15 +141,6 @@ export class RouterStore {
     this.navigationRefSubscription = navigationRef.addListener('state', (data) => {
       const state = data.data.state as ResultState;
 
-      if (!this.hasAttemptedToHideSplash) {
-        this.hasAttemptedToHideSplash = true;
-        // NOTE(EvanBacon): `navigationRef.isReady` is sometimes not true when state is called initially.
-        this.splashScreenAnimationFrame = requestAnimationFrame(() => {
-          // @ts-expect-error: This function is native-only and for internal-use only.
-          SplashScreen._internal_maybeHideAsync?.();
```

Comment:
Users will not be able to prevent white flash as the hooks for navigation container that are used here are not publicly available